test/drenv/tools/tools.py

```
@@ -0,0 +1,227 @@
+
+
+def download(url, target_path):
+    """
+    Download file from URL to target path.
+
+    Args:
+        url: URL to download from
+        target_path: Local file path to save to
+    """
+    logging.debug(f"Downloading {url} to {target_path}")
+    with urllib.request.urlopen(url) as response:
```

Ruff (0.15.17)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)